qutip/wigner.py

```diff
@@ -50,6 +50,179 @@
+        raise TypeError('Input state is not a valid operator.')
+
+    if psi.type == 'ket' or psi.type == 'bra':
+        rho = ket2dm(psi)
+    else:
+        rho = psi
+
+    sun = 2   # The order of the SU group
+
+    # calculate total number of particles in quantum state:
+    N = np.int32(np.log(np.shape(rho)[0]) / np.log(2 * j + 1))
+
+    theta = np.zeros((N, steps))
+    phi = np.zeros((N, steps))
+
+    for i in range(N):
+        theta[i, :] = np.linspace(0, np.pi, steps)
+        phi[i, :] = np.linspace(0, 2 * np.pi, steps)
+
+    theta, phi = _angle_slice(np.array(slicearray, dtype=str), theta, phi)
+
+    wigner = np.zeros((steps, steps))
+    if fullparity:
+        pari = _parity(sun**N, j)
+    else:
+        pari = _parity(sun, j)
+    for t in range(steps):
+        for p in range(steps):
+            wigner[t, p] = np.trace(rho.data @ _kernelsu2(
+                theta[:, t], phi[:, p], N, j, pari, fullparity))
+    return wigner
+
+
+def _parity(N, j):
+    """Private function to calculate the parity of the quantum system.
+    """
+    if j == 0.5:
+        pi = np.identity(N) - np.sqrt((N - 1) * N * (N + 1) / 2) * _lambda_f(N)
+        return pi / N
+    elif j > 0.5:
+        mult = np.int32(2 * j + 1)
+        matrix = np.zeros((mult, mult))
+        foo = np.ones(mult)
+        for n in np.arange(-j, j + 1, 1):
+            for l in np.arange(0, mult, 1):
+                foo[l] = (2 * l + 1) * qt.clebsch(j, l, j, n, 0, n)
+            matrix[np.int32(n + j), np.int32(n + j)] = np.sum(foo)
+        return matrix / mult
+
+
+def _lambda_f(N):
+    """Private function needed for the calculation of the parity.
+    """
+    matrix = np.sqrt(2 / (N * (N - 1))) * np.identity(N)
+    matrix[-1, -1] = - np.sqrt(2 * (N - 1) / N)
+    return matrix
+
+
+def _kernelsu2(theta, phi, N, j, parity, fullparity):
+    """Private function that calculates the kernel for the SU2 unitary group.
+    """
+    U = np.ones(1)
+    # calculate the total rotation matrix (tensor product for each particle):
+    for i in range(0, N):
+        U = np.kron(U, _rotation_matrix(theta[i], phi[i], j))
+    if not fullparity:
+        op_parity = parity   # The parity for a one particle system
+        for i in range(1, N):
+            parity = np.kron(parity, op_parity)
+    matrix = U @ parity @ U.conj().T
+    return matrix
+
+
+def _rotation_matrix(theta, phi, j, h=1):
+    """Private function to calculate the rotation operator for the SU2 kernel.
+    """
+    return la.expm(1j * phi * _j_z(j, h)) @ la.expm(1j * theta * _j_y(j, h))
+
+
+def _j_y(j, h=1):
+    """Private function to calculate the y angular momentum operator matrix.
+    """
+    return (_j_plus(j, h) - _j_minus(j, h)) / 2j
+
+
+def _j_z(j, h=1):
+    """Private function to calculate the z angular momentum operator matrix.
+    """
+    mult = np.int32(2 * j + 1)
+    matrix = np.zeros((mult, mult))
+    np.fill_diagonal(matrix, np.int32(h * np.linspace(-j, j, mult)))
```

Comment:
Here the `np.int32` round the 0.5 to 0. So `_j_z(0.5,1)` return an empty matrix. Removing it make the test passes.